api4/user.go

```diff
@@ -547,23 +546,26 @@ func searchUsers(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	searchOptions := map[string]bool{}
-	searchOptions[store.USER_SEARCH_OPTION_ALLOW_INACTIVE] = props.AllowInactive
+	if props.Limit <= 0 || props.Limit > model.USER_SEARCH_MAX_LIMIT {
+		c.SetInvalidParam("limit")
+		return
+	}
 
-	if !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
-		hideFullName := !c.App.Config().PrivacySettings.ShowFullName
-		hideEmail := !c.App.Config().PrivacySettings.ShowEmailAddress
-
-		if hideFullName && hideEmail {
-			searchOptions[store.USER_SEARCH_OPTION_NAMES_ONLY_NO_FULL_NAME] = true
-		} else if hideFullName {
-			searchOptions[store.USER_SEARCH_OPTION_ALL_NO_FULL_NAME] = true
-		} else if hideEmail {
-			searchOptions[store.USER_SEARCH_OPTION_NAMES_ONLY] = true
-		}
+	options := &model.UserSearchOptions{
+		IsAdmin:       c.IsSystemAdmin(),
+		AllowInactive: props.AllowInactive,
+		Limit:         props.Limit,
 	}
 
-	profiles, err := c.App.SearchUsers(props, searchOptions, c.IsSystemAdmin())
+	if c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
```

Comment:
why the difference in the options set here? I don't quite follow

Reply:
@enahum, just want to clarify the source of confusion. Are you asking about why `AllowEmails`/`AllowFullNames` is sometimes unilaterally `true` and sometimes tied to the `PrivacySettings`? Or about the difference between this function's option set and the `autocompleteUsers` below?

Reply:
Yes I’m not following why we unilaterally set them to true or to follow the privacy settings, I’m sure there is a reason I just don’t know what that reason is

Reply:
Typically, the server sanitizes the resulting user profiles for non-admins depending on the configured `PrivacySettings`. This code block mirrors that thinking except by filtering the allowed fields on which to search -- so admins would be allowed to match on email or full name, regardless of the privacy settings. 

I suppose this lines up with client-side expectations, given that admins "see" those fields unlike other users for which the resulting user profiles would have been sanitized.

Reply:
Thanks for the explanation

Reply:
Great addition. This is much clearer than the old version.